### ISSUE TITLE
Java 9 Support

### ISSUE THREAD

**Issue by mgardiner** [stoicflame/enunciate#742]
Java 9 has now been released and Oracle has announced that Java 8 end of public updates will be September 2018.  See http://www.oracle.com/technetwork/java/eol-135779.html.

Enunciate 2.9.1 is using internal Java APIs which was found by running jdeps -jdkinternals on the jars.

* enunciate-core - com.sun.tools.javac.api.JavacTool
* enunciate-java-json-client - com.sun.tools.javac.api.JavacTool
* enunciate-java-xml-client - com.sun.tools.javac.api.JavacTool

Under Java 8 these are warnings but under Java 9 they will not work.

Below is an example output of running jdeps -jdkinternals on enunciate-core:

Warning: JDK internal APIs are unsupported and private to JDK implementation that are
subject to be removed or changed incompatibly and could break your application.
Please modify your code to eliminate dependence on any JDK internal APIs.
For the most recent update on JDK internal API replacements, please check:
https://wiki.openjdk.java.net/display/JDK8/Java+Dependency+Analysis+Tool

JDK Internal API                         Suggested Replacement
----------------                         ---------------------
com.sun.tools.javac.api.JavacTool        Use javax.tools and javax.lang.model @since 1.6

There may be issues related to the versions of the dependencies Enunciate uses since they may not be Java 9 compatible as well.

**Comment by stoicflame** [comment]
This will need to be pushed to 2.11.

**Comment by stolp** [comment]
We are currently blocked to move to JDK 9 with our project by the enunciate dependency which we of course cannot afford to loose 🥇.

There will probably be others out there as well.

Not pushing, just a vote.

**Comment by stoicflame** [comment]
Agreed. It's a high priority.

**Comment by quibbleahr** [comment]
Our projects are also dependent on Enunciate, and we can't move to JDK 9 because of this issue. The last update was in October and JDK 10 is being released next month. Is there any word on when this will be fixed?
Thank you!

**Comment by stoicflame** [comment]
As of 44d8aaf Enunciate should work on Java 9.

You'll see some new annoying warnings, though:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by freemarker.ext.beans.BeansWrapper (file:/home/heatonra/.m2/repository/org/freemarker/freemarker/2.3.23/freemarker-2.3.23.jar) to method com.sun.tools.javac.util.Name.toString()
WARNING: Please consider reporting this to the maintainers of freemarker.ext.beans.BeansWrapper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

This is happening because Freemarker is using reflecting to get the string form of `com.sun.tools.javac.util.Name`. I'll probably need to fix this with some custom Freemarker fixes.

```
The POM for com.sun.xml.bind:jaxb-core:jar:2.2.11 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
The POM for com.sun.xml.bind:jaxb-impl:jar:2.2.11 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
```

This will be seen by maven users, and I have no idea how to fix it. Any suggestions would be appreciated.

**Comment by mgardiner** [comment]
Thanks for the update Ryan, there's a newer version of Freemarker available (2.3.28) which was just released 2018-04-04 that may resolve some of the reflection issues.

Also, many of the opensource Java libraries are specifying an automatic module name in the manifest to mark the artifact as compatible with Java 9 and newer (i.e. Java Platform Module System (JPMS)) until they support full JPMS via a module-info.class.

As far as the transitive dependency issue it may be worth deleting your local maven repo and rebuilding.

I will try to test these changes tomorrow under Java 9 or 10 as well.
